File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.DotNet/director/Chunks/RaysScriptNamesChunk.cs
using ProjectorRays.Common;
using ProjectorRays.Director;
using ProjectorRays.LingoDec;

namespace ProjectorRays.director.Chunks;

public class RaysScriptNamesChunk : RaysChunk
{
    public ScriptNames Names;

    public RaysScriptNamesChunk(RaysDirectorFile? dir) : base(dir, ChunkType.ScriptNamesChunk)
    {
        Names = new ScriptNames(dir?.Version ?? 0);
    }

    public override void Read(ReadStream stream)
    {
        Names.Read(stream);
    }

    public override void WriteJSON(RaysJSONWriter json)
    {
        json.StartObject();
        json.WriteField("namesCount", Names.Names.Count);
        json.WriteKey("names");
        json.StartArray();
        foreach (var n in Names.Names)
            json.WriteVal(n);
        json.EndArray();
        json.EndObject();
    }
}

File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.Python/director/chunks/rays_script_names_chunk.py
from .rays_chunk import RaysChunk, ChunkType
from ...common.stream import ReadStream

class ScriptNames:
    def __init__(self):
        self.names = []

class RaysScriptNamesChunk(RaysChunk):
    def __init__(self, dir=None):
        super().__init__(dir, ChunkType.ScriptNamesChunk)
        self.names = ScriptNames()

    def read(self, stream: ReadStream):
        while not stream.eof():
            length = stream.read_uint8()
            self.names.names.append(stream.read_string(length))
